torch/functional.py

```diff
@@ -326,6 +328,71 @@ def parse_subscript(n: int) -> str:
 
     return _VF.einsum(equation, operands)  # type: ignore[attr-defined]
 
+# Wrapper around _histogramdd and _histogramdd_bin_edges needed due to (Tensor, Tensor[]) return type.
+if TYPE_CHECKING:
+    # The JIT doesn't understand Union, so leave untyped for mypy
+    pass
+else:
+    def histogramdd(input: Tensor,
+                    bins: Union[List[Tensor], List[int]],
+                    range: Optional[List[float]] = None,
+                    weight: Optional[Tensor] = None,
+                    density: bool = False):
+        r"""
+        histogramdd(input, bins, *, range=None, weight=None, density=False, out=None) -> (Tensor, Tensor[])
+
+        Computes a histogram of the values in a tensor. Interprets the elements of an
+        input tensor with innermost dimension D as D-dimensional coordinates.
+
+        :attr:`bins` can be a sequence of D ints or a sequence of D 1D tensors.
+
+        If :attr:`bins` is a sequence of ints, it specifies the number of equal-width
+        bins in each dimension. By default, the lower and upper range of the bins is
+        determined by the minimum and maximum elements of the input tensor in the
+        corresponding dimension. The :attr:`range` argument can be provided to specify
+        ranges for the bins in each dimension.
+
+        If :attr:`bins` is a sequence of 1D tensors, it specifies the sequences of bin
+        edges, each including their rightmost edge. Each bin sequence should contain
+        an increasing sequence of at least 2 elements.
+
+        Args:
+            {input}
+            bins: int[] or Tensor[]. If int[], defines the number of equal-width bins
+                  in each dimension. If Tensor[], defines the sequences of bin edges,
+                  each including their rightmost edge.
+        Keyword args:
+            range (tuple of float): Defines the ranges of the bins in each dimension.
+            weight (Tensor): If provided, weight should have the same shape as input
+                             excluding its innermost dimension. Each D-dimensional
+                             coordinate in input contributes its associated weight
+                             towards its bin's result.
+            density (bool): If False, the result will contain the count (or total weight)
+                            in each bin. If True, the result is the value of the probability
+                            density function over the bins, normalized such that the integral
+                            over the range of the bins is 1.
+        Returns:
+            hist (Tensor): D-dimensional Tensor containing the values of the histogram.
+            bin_edges(Tensor[]): sequence of D 1D Tensors containing the edges of the histogram bins.
+
+        Example::
```

Comment:
It might be interesting to expand the examples and show how additional dims are just flattened, how coordinates with different values work, how a different number of bins can be selected per dim, and how range can be used to filter.